Error handling and debuging

samgun-6 opened the pull request:
# Related issue
- Fixes #37, Fixes #63 

# Proposed changes
- Change the arduino code so that it uses ultrasonic sensor
- Remove websocket implementation since the http request performance has been stabilized
- Refactoring codebase to use the string resource file
- Add application error handling class that handles runtime crashes

krasen86 committed:
Add runtime error handling for the android application

Add exception handler class for actives in order to restart the android application instead of crashing, add logic to stop the car once a runtime error occurs, improve the sending request logic so that it does not add unnecessary request to the queue.

 Issues:
 #63, #37

 Contributors:
 @samgun-6
 @krasen86

krasen86 committed:
Code refactoring and improve error handling logic

Add exception error message method in CarSelect, refactor the strings form the toast messages and move them to the values string file, update string values xml file with the new strings

 Issues:
#37, #60

 Contributors:
 @krasen86

krasen86 committed:
Add logic to handle lost of connection while in drive mode

Build upon the http request handling so that the application can handle lost to connection to the car while in drive mode(manual or/and voice)

 Issues:
#31

 Contributors:
 @krasen86

samgun-6 committed:
"Update logic in ManualControl and Remove error handling in HttpWebService for loss of connection"

samgun-6 committed:
Refactor code and Update Arduino Sketch

Description
Update healthRover.ino sketch with new hc-sr04 sensor implementation in order to improve the smartcar performance. Refactor and update the code base okHttpWebService

Issues: #37, #60, #63

Contributors:
@samgun-6
@krasen86

samgun-6 committed:
Add comments to newly added methods

Contributors:
@samgun-6
@krasen86

Enilo reviewed (commented):
I believe the naming convention we have been sticking to so far in the .xml files is with lower case letters and underscores to separate words. E.g carName would be car_name.

Also, as discussed on messenger; after we merge with our ui-refactoring the toast messages can be handled by another method to simplify the code.

Nice job guys, you have been working hard!

krasen86 committed:
Update sting values file

Change name stiling to be consistent with previous version.

 Contributors:
 @krasen86

krasen86 commented:
Updated the strings, we can update the toast messages when the method is ready.
